### PR TITLE
Migrate data generators to datasqrl-examples repository

### PR DESCRIPTION
**Pull request by nandorsoma**

### CODE REVIEW

File: data-generator/pom.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>sqrl-root</artifactId>
+        <groupId>run.sqrl</groupId>
+        <version>0.5.0-RC2-SNAPSHOT</version>
```
Comment (nandorsoma) [inline data-generator/pom.xml:10]: Do we want to use a common parent to ensure that our examples are using the same dependencies that the main project is using?

File: data-generator/pom.xml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    <artifactId>data-generator</artifactId>
+
+    <properties>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <dependencies>
+<!--        <dependency>-->
```
Comment (nandorsoma) [inline data-generator/pom.xml:22]: It seems like the full module is not needed, I've extracted the transitive utility dependencies from it below. Is it good like that? If so, I'll remove the commented out section.

File: data-generator/pom.xml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    <parent>
+        <artifactId>sqrl-root</artifactId>
+        <groupId>run.sqrl</groupId>
+        <version>0.5.0-RC2-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>data-generator</artifactId>
+
+    <properties>
+        <maven.compiler.source>11</maven.compiler.source>
```
Comment (nandorsoma) [inline data-generator/pom.xml:16]: Currently lombok prevents upgrading to 21. If we drop the root pom, we can upgrade to 21 here.

File: data-generator/pom.xml (hunk 1 above)
Comment (mbroecheler) [reply data-generator/pom.xml:10]: Yes, let's try to make this repository standalone and not depend on the sqrl project. What classes are we pulling in from sqrl-root that we need here?

File: data-generator/pom.xml (hunk 2 above)
Comment (mbroecheler) [reply data-generator/pom.xml:22]: Great! It looks like we don't need any classes from the `sqrl` project, right? Let's make this standalone then.

**Review by mbroecheler — APPROVED** [review]
Looks good. Did you try running the tests in TestDataGenerators and they still work? Just as a smoke test. Thanks for getting this migrated.

**Pull request merged**